examples/app-service-asp-net/app.tf

```diff
@@ -0,0 +1,65 @@
+# Configure the Microsoft Azure Provider
+provider "azurerm" {
+  subscription_id = "${var.subscription_id}"
+  client_id       = "${var.client_id}"
+  client_secret   = "${var.client_secret}"
+  tenant_id       = "${var.tenant_id}"
+}
+
+resource "azurerm_resource_group" "g1" {
+  name     = "{$var.groupName}"
+  location = "${var.location}"
+}
+
+resource "azurerm_app_service_plan" "plan0" {
+  name                = "service-plan0"
+  location            = "${var.location}"
+  resource_group_name = "${azurerm_resource_group.g1.name}"
+
+  sku {
+    tier = "Basic" # Basic | Standard | ...
+    size = "B1"    # B1 | S1 | ...
+  }
+}
+
+# underscores not supported as app_service name -> if not: you will receive error 400
+
+resource "azurerm_app_service" "common_service" {
+  name                = "${var.webName}"
+  location            = "${var.location}"
+  resource_group_name = "${azurerm_resource_group.g1.name}"
+  app_service_plan_id = "${azurerm_app_service_plan.plan0.id}"
+
+  site_config {
+    dotnet_framework_version = "v4.0"
+    remote_debugging_enabled = true
+    remote_debugging_version = "VS2015"
+  }
+
+  # app_settings {
+  #   "SOME_KEY" = "some-value"
+  # }
+  # connection_string {
+  #   name  = "Database"
+  #   type  = "SQLServer"
+  #   value = "Server=some-server.mydomain.com;Integrated Security=SSPI"
+  # }
+
+  provisioner "local-exec" {
+    command = "curl -k -u ${var.deploy_user}:${var.deploy_pass} -X PUT --data-binary @${var.deployZipFile} https://${azurerm_app_service.common_service.name}.scm.azurewebsites.net/api/zip/site/wwwroot/"
```

Comment:
out of interest, how are these values set? From memory these need to be set either in the portal/through the CLI before use, given Terraform currently doesn't support them and they're disabled by default?

Reply:
Thanks for the review Tom.
Yes, I didn't find a way yet to setup such deploy Azure credentials in a CLI/programmatic way.
As far as I know, them should be setup once in the Azure Portal per Azure Account.
Probably MS Azure guys can help with it.

Reply:
hey @pjmolina 

Apologies for the delayed response here!

In order for us to be able to ship this PR (without making the user having to navigate to the portal to enter the credentials) - I've pushed a commit to remove the publishing credentials snippet from the sample for the moment. Once support for configuring the App Service Publish Credentials is available through Terraform we can look into adding this back in - however for the moment this feels like the best compromise to ship this.

I've also opened #612 which exports the Default Hostname for the App Service - such that we can determine what it is (since `azurewebsites.net` is only valid for Azure Public and not China/Germany/Government) - so once that's merged this PR should be good to merge :)

Hope that makes sense and you don't mind?

Thanks!

Reply:
Looks great @tombuildsstuff !

Reply:
@pjmolina totally get that - we'll look to add it back in to the example once support for configuring the credentials is added to the Provider :)